### PR TITLE
Bugfix for NewRange doesn't init maxBytes Parameter

### PR DESCRIPTION
**Pull request by Frank-Jin**
Range.maxBytes is set at BootstrapRange and stored in the rocksdb at start-up stage, but NewRange doesn't read this parameter from the rocksdb, unless runSetZone configure this parameter and synchronize to cluster by gossip.

### CODE REVIEW

**Comment by tschottdorf** [comment]
Isn't this done at the store level?

```
func (s *Store) setRangesMaxBytes(zoneMap PrefixConfigMap) {
	s.mu.Lock()
	defer s.mu.Unlock()
	zone := zoneMap[0].Config.(*proto.ZoneConfig)
	idx := 0
	for _, rng := range s.ranges {
		if idx < len(zoneMap)-1 && !rng.Desc().StartKey.Less(zoneMap[idx+1].Prefix) {
			idx++
			zone = zoneMap[idx].Config.(*proto.ZoneConfig)
		}
		rng.SetMaxBytes(zone.RangeMaxBytes)
	}
}
```

**Comment by Frank-Jin** [comment]
setRangesMaxBytes is called by configGossipUpdate that is triggered when runSetZone changes the configuration.  If runSetZone doesn't run, Range.maxBytes can't be initialized.

**Comment by tschottdorf** [comment]
I see. It looks like we never put the zone configuration into gossip when booting up the node (I might have missed it), so the correct fix would be to make sure that happens.
Gossip already makes sure that when you add a callback, it is invoked automatically (whether or not there's data there), so that should be all that needs to be fixed.

**Comment by Frank-Jin** [comment]
BootstrapRange:
// Zone config.
	zoneConfig := &proto.ZoneConfig{
		ReplicaAttrs: []proto.Attributes{
			{},
			{},
			{},
		},
		RangeMinBytes: 1048576,
		RangeMaxBytes: 67108864,
		GC: &proto.GCPolicy{
			TTLSeconds: 24 * 60 * 60, // 1 day
		},
	}
	key = engine.MakeKey(engine.KeyConfigZonePrefix, engine.KeyMin)
	if err := engine.MVCCPutProto(batch, ms, key, now, nil, zoneConfig); err != nil {
		return err
	}
BootstrapRange initialized the maxBytes parameter, but isn't used later

File: storage/range.go [hunk 1]
```diff
@@ -208,7 +208,17 @@ func NewRange(desc *proto.RangeDescriptor, rm RangeManager) (*Range, error) {
 	}
 	r.SetDesc(desc)
 
-	err := r.loadLastIndex()
+	var zoneConfig *proto.ZoneConfig
```
Comment (spencerkimball) [inline storage/range.go]: This value is set by the store very efficiently for all of the ranges in setRangeMaxBytes. Doing it here would be quite inefficient (keep in mind that there may be a million ranges in a single store, so this means every range at startup would have to call through to read the zone config).

Also, you're only getting the default zone config here, which won't apply to all ranges. Further, reading the zone config directly from the engine only will work if the store contains the range with that zone config. And one more thing--reading this with proto.ZeroTimestamp won't work as zone configs are stored using MVCC. You'd need to use the current time or else you'd end up missing the written value as the timestamp would be earlier than any written value.

**Comment by Frank-Jin** [comment]
You are right, I'd better use gossip to trigger the parameter to be initialized.

**Comment by spencerkimball** [comment]
You shouldn't have to put the zone config into gossip. The range which holds the zone config info will gossip it at startup. @tschottdorf, look at the code in Range.start() and Range.maybeGossipConfigs()

**Comment by tschottdorf** [comment]
Ah, my grep-fu wasn't good enough for those. Then there's nothing to be done, is there?

**Comment by spencerkimball** [comment]
I did notice another bug--will send you a PR.

**Comment by tschottdorf** [comment]
@Frank-Jin looks like all of this should be working already. If you opened this PR because it didn't work for you, please let us know (along with some details) and we'll take another look.

**Pull request closed without merging**